### DISCUSSION TITLE
Custom publish based on artifact type

### DISCUSSION THREAD

**Discussion by ryancurrah** [goreleaser/goreleaser#4396] [Q&A] [answered]
We are using Go Releaser Pro and currently do not use it to publish but are investigating the potential to use it

The following reasons are why we have not used Go Releaser for publishing artifacts...

1. We want to maintain the publish configuration globally instead of in each repository.
2. We run the build step and publish step a separately.

To solve for problem one we would like to use the custom publisher with jfrog cli and put the custom publish steps in our global Go Releaser config file to upload debians, binaries, docker images. 

We would like to have a custom publisher configuration per artifact type. So I was wondering if there is a way to filter or only run a custom publish on a specific artifact type?

**Comment by caarlos0** [comment]
not right now, you could though run a shell script, and filter there, although its less clean.

def something to improve on goreleaser's side!

**Reply by ryancurrah** [reply]
That could be a good hack for now. Can you use custom publishers with Docker Images?

**Reply by caarlos0** [reply]
you should be able to, yes

**Comment by caarlos0** [comment, marked as answer]
v2.5 will allow an `if` filter in custom publishers

https://goreleaser.com/customization/publishers/
